modules/Microsoft.Cache/redis/deploy.bicep

```diff
@@ -0,0 +1,262 @@
+    sku: {
+      capacity: capacity
+      family: family
+      name: skuName
+    }
+    staticIP: !empty(staticIP) ? staticIP : null
+    subnetId: !empty(subnetId) ? subnetId : null
+    tenantSettings: tenantSettings
+  }
+  zones: skuName == 'Premium' ? pickZones('Microsoft.Cache', 'redis', location, 1) : null
+}
+
+resource redisCache_lock 'Microsoft.Authorization/locks@2017-04-01' = if (!empty(lock)) {
+  name: '${redisCache.name}-${lock}-lock'
+  properties: {
+    level: any(lock)
+    notes: lock == 'CanNotDelete' ? 'Cannot delete resource or child resources.' : 'Cannot modify the resource or child resources.'
+  }
+  scope: redisCache
+}
+
+resource redisCache_diagnosticSettings 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = if (!empty(diagnosticStorageAccountId) || !empty(diagnosticWorkspaceId) || !empty(diagnosticEventHubAuthorizationRuleId) || !empty(diagnosticEventHubName)) {
+  name: diagnosticSettingsName
+  properties: {
+    storageAccountId: empty(diagnosticStorageAccountId) ? null : diagnosticStorageAccountId
+    workspaceId: empty(diagnosticWorkspaceId) ? null : diagnosticWorkspaceId
+    eventHubAuthorizationRuleId: empty(diagnosticEventHubAuthorizationRuleId) ? null : diagnosticEventHubAuthorizationRuleId
+    eventHubName: empty(diagnosticEventHubName) ? null : diagnosticEventHubName
+    metrics: empty(diagnosticStorageAccountId) && empty(diagnosticWorkspaceId) && empty(diagnosticEventHubAuthorizationRuleId) && empty(diagnosticEventHubName) ? null : diagnosticsMetrics
+    logs: empty(diagnosticStorageAccountId) && empty(diagnosticWorkspaceId) && empty(diagnosticEventHubAuthorizationRuleId) && empty(diagnosticEventHubName) ? null : diagnosticsLogs
+  }
+  scope: redisCache
+}
+
+module redisCache_rbac '.bicep/nested_rbac.bicep' = [for (roleAssignment, index) in roleAssignments: {
+  name: '${uniqueString(deployment().name, location)}-AppGateway-Rbac-${index}'
+  params: {
+    description: contains(roleAssignment, 'description') ? roleAssignment.description : ''
+    principalIds: roleAssignment.principalIds
+    principalType: contains(roleAssignment, 'principalType') ? roleAssignment.principalType : ''
+    roleDefinitionIdOrName: roleAssignment.roleDefinitionIdOrName
+    resourceId: redisCache.id
+  }
+}]
+
+@description('The resource name.')
+output name string = redisCache.name
+
+@description('The resource id.')
+output resourceId string = redisCache.id
+
+@description('The name of the resource group the Redis cache was created in.')
+output resourceGroupName string = resourceGroup().name
+
+@description('Redis host name.')
+output hostName string = redisCache.properties.hostName
+
+@description('Redis SSL port.')
+output sslPort int = redisCache.properties.sslPort
+
+@description('The full resource ID of a subnet in a virtual network where the Redis cache was deployed in.')
+output subnetId string = !empty(subnetId) ? redisCache.properties.subnetId : ''
```

Comment:
Are all these additional outputs intended?

Reply:
Yes they are intended. I had usecases for these, and added them. As per design of the CARML modules, we have the base outputs plus the resource specific ones that makes sense and that don't expose security relevant information like access keys. That's at least how I understood it, maybe @MrMCake can elaborate :)